presto-native-execution/src/test/java/com/facebook/presto/nativeworker/ContainerQueryRunner.java

```diff
@@ -0,0 +1,384 @@
+                }
+                else {
+                    currentField.append(ch);
+                }
+            }
+            if (insideQuotes) {
+                currentField.append('\n'); // Add newline for multiline fields
+            }
+            else {
+                currentRow.add(currentField.toString());
+                currentField.setLength(0); // Clear the current field
+                allRows.add(currentRow.toArray(new String[0]));
+                currentRow.clear();
+            }
+        }
+        if (!currentRow.isEmpty()) {
+            currentRow.add(currentField.toString());
+            allRows.add(currentRow.toArray(new String[0]));
+        }
+        return allRows;
+    }
+
+    private static Type inferType(List<String> values)
+    {
+        boolean isBigint = true;
+        boolean isDouble = true;
+        boolean isBoolean = true;
+
+        for (String value : values) {
+            if (!value.matches("^-?\\d+$")) {
+                isBigint = false;
+            }
+            if (!value.matches("^-?\\d+\\.\\d+$")) {
+                isDouble = false;
+            }
+            if (!value.equalsIgnoreCase("true") && !value.equalsIgnoreCase("false")) {
+                isBoolean = false;
+            }
+        }
+
+        if (isBigint) {
+            return BigintType.BIGINT;
+        }
+        else if (isDouble) {
+            return DoubleType.DOUBLE;
+        }
+        else if (isBoolean) {
+            return BooleanType.BOOLEAN;
+        }
+        else {
+            return VarcharType.VARCHAR;
+        }
+    }
+
+    private static Object convertToType(String value, Type type)
+    {
+        if (value.isEmpty()) {
+            return null;
+        }
+        if (type.equals(VarcharType.VARCHAR)) {
+            return value;
+        }
+        else if (type.equals(BigintType.BIGINT)) {
+            return Long.parseLong(value);
+        }
+        else if (type.equals(DoubleType.DOUBLE)) {
+            return Double.parseDouble(value);
+        }
+        else if (type.equals(BooleanType.BOOLEAN)) {
+            return Boolean.parseBoolean(value);
+        }
+        else {
+            throw new IllegalArgumentException("Unsupported type: " + type);
+        }
+    }
+
+    public Container.ExecResult executeQuery(String sql)
+    {
+        String[] command = {
+                "/opt/presto-cli",
+                "--server",
+                "presto-coordinator:8081",
+                "--catalog",
+                "tpch",
+                "--schema",
+                "tiny",
+                "--execute",
+                sql
+        };
+
+        Container.ExecResult execResult = null;
+        try {
+            execResult = coordinator.execInContainer(command);
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
Instead of rethrowing, can we just fail?

Reply:
Updated the code as per your discussion.